cirq-core/cirq/transformers/transformer_api.py

```diff
@@ -291,22 +309,36 @@ def method_with_logging(
         func = cls_or_func
 
         @functools.wraps(func)
-        def func_with_logging(
-            circuit: 'cirq.AbstractCircuit', context: TransformerContext
-        ) -> 'cirq.AbstractCircuit':
-            return _transform_and_log(func, func.__name__, circuit, context)
+        def func_with_logging(circuit, **kwargs) -> 'cirq.AbstractCircuit':
+            return _transform_and_log(
+                func,
+                func.__name__,
+                circuit,
+                _get_context(func, f'{func.__name__}()', **kwargs),
+                **kwargs,
+            )
 
         return func_with_logging
 
 
+def _get_context(func, name, **kwargs):
+    if 'context' in kwargs:
+        return kwargs['context']
+    sig = inspect.signature(func)
+    default_context = sig.parameters["context"].default
+    assert (
+        default_context != inspect.Parameter.empty
+    ), "`context` argument must have a default value specified."
```

Comment:
nit: I'd suggest raising a different error, such as `ValueError`, rather than using `assert`.

Reply:
The Transformer protocol now includes a default value `None` for the context, hence mypy type checking would fail if a user uses `@cirq.transformer` on a method which does not provide a default value for `context`. That's why I've used an assert instead of a value error.